src/peft/tuners/tuners_utils.py

```diff
@@ -483,30 +481,41 @@ def check_target_module_exists(config, key: str) -> bool | re.Match[str] | None:
         None if no match found
     """
     if isinstance(config.target_modules, str):
-        target_module_found = re.fullmatch(config.target_modules, key)
+        target_module_found = (config.target_modules == key) or re.fullmatch(config.target_modules, key)
```

Comment:
```suggestion
        target_module_found = re.fullmatch(config.target_modules, key)
```
those are equivalent no?

Reply:
Thanks for the suggestion, indeed the suggestion is the previous code. I change the code because I see `target_module_found = key in config.target_modules or ...` when the type of `target_modules` is list.  And if the `target_modules` is regex it will be fine, but if the `target_modules` is just the complete layer name and contains special characters in regex, it may cause problem. But for the common layer name like `foo.bar.1.baz`, it doesn't matter, although `.` is special in regex.
  In a word, `target_module_found = re.fullmatch(config.target_modules, key)` should be fine in almost all the cases, but it doesn't hurt to add `(config.target_modules == key) or `.

Reply:
ok looks good to me, thanks for explaining!